src/charm.py

```diff
@@ -146,12 +158,15 @@ def _render_conf_file(self) -> None:
         rendered = template.render(
             mappers_path=str(self._mappers_dir_path),
             identity_schema_file_path=self._identity_schema_file_path,
-            default_browser_return_url="http://127.0.0.1:9999/",
-            login_ui_url="http://localhost:4455/login",
+            default_browser_return_url=self.config.get("login_ui_url"),
+            public_base_url=self._domain_url,
+            login_ui_url=join(self.config.get("login_ui_url"), "login"),
+            error_ui_url=join(self.config.get("login_ui_url"), "oidc_error"),
             oidc_providers=self.external_provider.get_providers(),
             available_mappers=self._get_available_mappers,
-            registration_ui_url="http://127.0.0.1:9999/registration",
+            registration_ui_url=join(self.config.get("login_ui_url"), "registration"),
             db_info=self._get_database_relation_info(),
+            oauth2_provider_url=self._get_hydra_endpoint_info(),
```

Comment:
Here we sould simetimes render a template with an empty `oauth2_provider_url`, is this normal?

Reply:
If `oauth2_provider_url` is empty, jinja template will not add it to the config:
```
{%- if oauth2_provider_url %}
oauth2_provider:
    url: {{ oauth2_provider_url }}
{%- endif %}
```